File: src/tensorflow/ops/io/bigquery/bigquery.cc
#include "tensorflow/cc/ops/const_op.h"
#include "bigquery.h"

BigQueryReader::BigQueryReader(tensorflow::Scope& scope, 
           tensorflow::string container, 
           tensorflow::string shared_name, 
           tensorflow::string project_id, 
           tensorflow::string dataset_id, 
           tensorflow::string table_id, 
           tensorflow::gtl::ArraySlice<tensorflow::string> columns, 
           tensorflow::string test_end_point, 
           int64_t timestamp_millis) {
      ::tensorflow::Node *ret;
      const auto unique_name = scope.GetUniqueNameForOp("BigQueryReader");
      auto builder = ::tensorflow::NodeBuilder(unique_name, "BigQueryReader")
                                   .Attr("container", container)
                                   .Attr("shared_name", shared_name)
                                   .Attr("project_id", project_id)
                                   .Attr("dataset_id", dataset_id)
                                   .Attr("table_id", table_id)
                                   .Attr("columns", columns)
                                   .Attr("test_end_point", test_end_point)
                                   .Attr("timestamp_millis", timestamp_millis)
      ;
      scope.UpdateBuilder(&builder);
      scope.UpdateStatus(builder.Finalize(scope.graph(), &ret));
      if (!scope.ok()) return;
      scope.UpdateStatus(scope.DoShapeInference(ret));
      this->operation = ::tensorflow::Operation(ret);
      this->output = tensorflow::Output(ret, 0);
}

GenerateBigQueryReaderPartitions::GenerateBigQueryReaderPartitions(tensorflow::Scope& scope, 
           tensorflow::string project_id, 
           tensorflow::string dataset_id, 
           tensorflow::string table_id, 
           tensorflow::gtl::ArraySlice<tensorflow::string> columns, 
           tensorflow::string test_end_point, 
           int64_t timestamp_millis, 
           int64_t num_partitions) {
      ::tensorflow::Node *ret;
      const auto unique_name = scope.GetUniqueNameForOp("GenerateBigQueryReaderPartitions");
      auto builder = ::tensorflow::NodeBuilder(unique_name, "GenerateBigQueryReaderPartitions")
                                   .Attr("project_id", project_id)
                                   .Attr("dataset_id", dataset_id)
                                   .Attr("table_id", table_id)
                                   .Attr("columns", columns)
                                   .Attr("test_end_point", test_end_point)
                                   .Attr("timestamp_millis", timestamp_millis)
                                   .Attr("num_partitions", num_partitions)
      ;
      scope.UpdateBuilder(&builder);
      scope.UpdateStatus(builder.Finalize(scope.graph(), &ret));
      if (!scope.ok()) return;
      scope.UpdateStatus(scope.DoShapeInference(ret));
      this->operation = ::tensorflow::Operation(ret);
      this->output = tensorflow::Output(ret, 0);
}

File: src/tensorflow/ops/io/bigquery.nim
import ../../core

{.compile:"bigquery/bigquery.cc".}


type BigQueryReader* {.header:"/Users/martin//.nimble/pkgs/tensorflow-0.1.0/tensorflow/ops/io/bigquery/bigquery.h", importcpp:"BigQueryReader/*'0*/".} = object
  operation*: Operation[ostring]
  output*: ostring

proc iibigQueryReader(scope: Scope,
                     container: cstring,
                     shared_name: cstring,
                     project_id: cstring,
                     dataset_id: cstring,
                     table_id: cstring,
                     columns: ArraySlice[cstring],
                     test_end_point: cstring,
                     timestamp_millis: int64): BigQueryReader {.header:"/Users/martin//.nimble/pkgs/tensorflow-0.1.0/tensorflow/ops/io/bigquery/bigquery.h", importcpp:"BigQueryReader(*#, tensorflow::string(#), tensorflow::string(#), tensorflow::string(#), tensorflow::string(#), tensorflow::string(#), #, tensorflow::string(#), #)", constructor.}

proc bigQueryReader*(scope: Scope,
                     container: cstring,
                     shared_name: cstring,
                     project_id: cstring,
                     dataset_id: cstring,
                     table_id: cstring,
                     columns: openArray[cstring],
                     test_end_point: cstring,
                     timestamp_millis: int64 = 0.int): BigQueryReader =
  return iibigQueryReader(scope,
                          container,
                          shared_name,
                          project_id,
                          dataset_id,
                          table_id,
                          newArraySlice(columns),
                          test_end_point,
                          timestamp_millis)

converter bigQueryReaderToOut*(op: BigQueryReader): ostring {.inline.} = return op.output


type GenerateBigQueryReaderPartitions* {.header:"/Users/martin//.nimble/pkgs/tensorflow-0.1.0/tensorflow/ops/io/bigquery/bigquery.h", importcpp:"GenerateBigQueryReaderPartitions/*'0*/".} = object
  operation*: Operation[ostring]
  output*: ostring

proc iigenerateBigQueryReaderPartitions(scope: Scope,
                                       project_id: cstring,
                                       dataset_id: cstring,
                                       table_id: cstring,
                                       columns: ArraySlice[cstring],
                                       test_end_point: cstring,
                                       timestamp_millis: int64,
                                       num_partitions: int64): GenerateBigQueryReaderPartitions {.header:"/Users/martin//.nimble/pkgs/tensorflow-0.1.0/tensorflow/ops/io/bigquery/bigquery.h", importcpp:"GenerateBigQueryReaderPartitions(*#, tensorflow::string(#), tensorflow::string(#), tensorflow::string(#), #, tensorflow::string(#), #, #)", constructor.}

proc generateBigQueryReaderPartitions*(scope: Scope,
                                       project_id: cstring,
                                       dataset_id: cstring,
                                       table_id: cstring,
                                       columns: openArray[cstring],
                                       test_end_point: cstring,
                                       timestamp_millis: int64 = 0.int,
                                       num_partitions: int64 = 0.int): GenerateBigQueryReaderPartitions =
  return iigenerateBigQueryReaderPartitions(scope,
                                            project_id,
                                            dataset_id,
                                            table_id,
                                            newArraySlice(columns),
                                            test_end_point,
                                            timestamp_millis,
                                            num_partitions)

converter generateBigQueryReaderPartitionsToOut*(op: GenerateBigQueryReaderPartitions): ostring {.inline.} = return op.output

    # This op should not be used directly by clients. Instead, the
    # bigquery_reader_ops.py file defines a clean interface to the reader.

